andela/boondocks-bn-frontend
# #171167298 Two factor authentication

#### What does this PR do?
Adds an extra layor of security upon user login (2 Factor Authentication)
#### Description of Task to be completed?
- adds a settings page where user can set method of 2FA to use or disable it
- adds a 2fa page after login when the user has activated one of the methods of 2fa (using sms or authenticator app like Google Authenticator)
#### How should this be manually tested?
- Clone the branch
- Run `npm run watch` to start the project
- Log in to the application and navigate to `settings` page
- Go to `Two factor authentication` link on the sidebar
- And activate one of the methods
- Logout and login, the user should be redirected to the page to validate the token
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#171167298](https://www.pivotaltracker.com/story/show/171206721)
#### Screenshots (if appropriate)

1. Setup first page (Setup)
![image](https://user-images.githubusercontent.com/3137192/74903385-6e1e0780-53b1-11ea-9568-0c2a4af2d2b8.png)

2. Setup scan page (Verification)
![image](https://user-images.githubusercontent.com/3137192/74903456-a3c2f080-53b1-11ea-8179-9997d45a0920.png)


3. (Login FA SMS)
![image](https://user-images.githubusercontent.com/3137192/74903650-42e7e800-53b2-11ea-8e46-975286d0e42e.png)

4. (Login FA Authenticator)
![image](https://user-images.githubusercontent.com/3137192/74903707-71fe5980-53b2-11ea-90d6-4bbcf905d89e.png)


#### Questions:
N/A
